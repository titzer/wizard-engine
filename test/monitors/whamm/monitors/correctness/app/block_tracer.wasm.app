../../../branch_monitor0.wasm
