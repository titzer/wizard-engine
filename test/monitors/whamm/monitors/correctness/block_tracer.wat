(module
  (func $putc (import "wizeng" "putc") (param i32))
  (func $puti (import "wizeng" "puti") (param i32))
  
  (func (export "wasm:block:start(fid,pc)") (param i32 i32)
    (call $putc (i32.const 115))
    (call $puti (local.get 0))
    (call $putc (i32.const 43))
    (call $puti (local.get 1))
    (call $putc (i32.const 32))
  )
  (func (export "wasm:block:end(fid,pc,instr_count)") (param i32 i32 i32)
    (call $putc (i32.const 101))
    (call $puti (local.get 0))
    (call $putc (i32.const 43))
    (call $puti (local.get 1))
    (call $putc (i32.const 44))
    (call $puti (local.get 2))
    (call $putc (i32.const 10))
  )
)
