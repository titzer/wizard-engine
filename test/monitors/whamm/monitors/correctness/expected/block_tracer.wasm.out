s0+0 e0+5,3
s0+7 e0+18,7
s0+7 e0+18,7
s0+7 e0+18,7
s0+7 e0+18,7
s0+7 e0+18,7
s0+20 e0+21,2
